pkg/operator/resource/resourceapply/json_patch_helpers.go

```diff
@@ -68,3 +69,26 @@ func JSONPatchSecretNoError(original, modified *corev1.Secret) string {
 
 	return JSONPatchNoError(safeOriginal, safeModified)
 }
+
+// JSONPatchRouteNoError work similarly to JSONPatchNoError but removes TLS Key
+func JSONPatchRouteNoError(original, modified *routev1.Route) string {
+	if original == nil {
+		return "original object is nil"
+	}
+	if modified == nil {
+		return "modified object is nil"
+	}
+
+	safeModified := modified.DeepCopy()
+	safeOriginal := original.DeepCopy()
+
+	if safeOriginal.Spec.TLS.Key != "" {
+		safeOriginal.Spec.TLS.Key = "MASKED"
+	}
+
+	if safeModified.Spec.TLS.Key != "" {
+		safeModified.Spec.TLS.Key = "MASKED"
+	}
```

Comment:
compare them and somehow mark them as changed if they are different

Reply:
Done